zebra-chain/src/orchard/commitment.rs

```diff
@@ -157,20 +144,18 @@ impl NoteCommitment {
         s.extend(rho_bytes);
         s.extend(psi_bytes);
 
-        let rcm = CommitmentRandomness(generate_trapdoor(csprng));
+        let rcm = CommitmentRandomness::from(note.rseed);
 
-        Some((
-            rcm,
-            NoteCommitment::from(
-                sinsemilla_commit(rcm.0, b"z.cash:Orchard-NoteCommit", &s)
-                    .expect("valid orchard note commitment, not ⊥ "),
-            ),
+        Some(NoteCommitment::from(
+            sinsemilla_commit(rcm.0, b"z.cash:Orchard-NoteCommit", &s)
+                .expect("valid orchard note commitment, not ⊥ "),
         ))
     }
 
     /// Hash Extractor for Pallas
     ///
     /// https://zips.z.cash/protocol/nu5.pdf#concreteextractorpallas
+    // TODO: dedupe with the other extract_x()
```

Comment:
Let's open a ticket?

Reply:
There is an extract_x function in orchard/sinsemilla.rs that does the same thing (implements Extract_P from the spec), this one should at minimum call that one, or maybe go away entirely in favor of that one

Reply:
But yeah either way we can open a ticket

Reply:
Got it, thanks! It was simple enough to do now, in c7af630505
(it's a bit inefficient since it ends up converting affine -> projective -> affine, but I think that's not important right now)